src/socket.rs

```diff
@@ -166,26 +172,102 @@ mod linux {
     }
 }
 
-#[cfg(target_os = "linux")]
-pub async fn relay(
-    downstream: &mut tokio::net::TcpStream,
-    upstream: &mut tokio::net::TcpStream,
-) -> Result<(u64, u64), Error> {
-    const EINVAL: i32 = 22;
-
-    match realm_io::bidi_zero_copy(downstream, upstream).await {
-        Ok(d) => Ok(d),
-        Err(ref e) if e.raw_os_error().map_or(false, |ec| ec == EINVAL) => {
-            tokio::io::copy_bidirectional(downstream, upstream).await
-        }
-        Err(e) => Err(e),
+// TLS record size max is 16k. But we also have a H2 frame header, so leave a bit of room for that.
+const BUFFER_SIZE: usize = 16_384 - 64;
+
+pub async fn copy_bidirectional<A, B>(
+    downstream: &mut A,
+    upstream: &mut B,
```

Comment:
We should be able to drop `realm_io` as a top-level dep, this was the only reference IIRC.